### DISCUSSION TITLE
Favorites

### DISCUSSION THREAD

**Discussion by mathieugoodman** [vicwomg/pikaraoke#208] [Ideas]
Hello!

I'm just full of ideas today, aren't I? 😉 

It would be nice to have a way to mark songs as a Favorite, so that if a singer sings it regularly, it can be found quickly without having to search for it. Ideally, it would support marking both local songs and youtube songs as favorites.

FYI I do have some basic programming skills so I could try to help with some of these if there's interest.

**Comment by charlie-mtl** [comment]
Hi team,
Even better, multiple sharable playlists per user.

Thank you,
Charlie

**Comment by RecursiveFunctions** [comment]
is this something that can be done with cookies?

we could add a favorite button next to each song and also display the option next to the song currently playing in case someone sings a song you like and want to add to your favorites.
```
<div class="song" data-song-id="{{ song.id }}">
   <h2>{{ song.title }}</h2>
   <button class="favorite-btn">❤️</button>
</div>
```
```
@app.route('/favorite/<song_id>')
def favorite(song_id):
   favorites = request.cookies.get('favorites', '')
   if song_id in favorites.split(','):
       favorites = ','.join([fav for fav in favorites.split(',') if fav != song_id])
   else:
       favorites += ',' + song_id
   resp = make_response('Updated favorites')
   resp.set_cookie('favorites', favorites, max_age=60*60*24*365*1000)
   return resp
```
just some example code. I'm spitballin here.



--
Or save it as a file in a subdirectory for users in pikaraoke. 

Users could create an account if they want a persistent list of their favorite songs.
